ibm/data_source_ibm_pi_placement_group.go

```diff
@@ -0,0 +1,77 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	st "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/helpers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceIBMPIPlacementGroup() *schema.Resource {
+
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIPlacementGroupRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			helpers.PIPlacementGroupName: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			helpers.PICloudInstanceId: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"members": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceIBMPIPlacementGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := meta.(ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
+	placementGroupName := d.Get(helpers.PIPlacementGroupName).(string)
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, cloudInstanceID)
+
+	response, err := client.Get(placementGroupName)
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.Errorf("Failed to get the placement group %v", err)
```

Comment:
No need for adding the message here as it is already part of err created by the client.

Reply:
Done.